src/V3LinkInc.cpp

```diff
@@ -201,6 +202,17 @@ class LinkIncVisitor final : public VNVisitor {
     void prepost_stmt_visit(AstNodeTriop* nodep) {
         iterateChildren(nodep);
 
+        // Currently we can't reference the target, so we just copy the AST both for read and
+        // write, but doing so would double any side-effects, so as a safety measure all
+        // statements which could have side-effects are banned at the moment.
+        if (MutatorAstSet::contains(nodep->rhsp())) {
```

Comment:
```suggestion
        if (!isTreePureRecurse(nodep->rhsp())) {
```

And see V3Const for that function.

Reply:
Thanks! I was aware of `isPure` method, but I guess it doesn't check the tree recursively, as it didn't work in place of this check. `isTreePureRecurse` looks like exactly what I've been looking for.